solutions/main.tf

```diff
@@ -303,10 +303,117 @@ resource "aws_instance" "application" {
   vpc_security_group_ids = ["${aws_security_group.peekops_main_app.id}"]
   subnet_id = "${aws_subnet.peekops_main_app_a.id}"
   tags {
-     Name = "peekops_application"
+     Name = "peekops_application_a"
+  }
+  provisioner "local-exec" {
```

Comment:
You can use any tool you like: 
- Ansible + Vagrant
- Puppet + Vagrant
- Chef + Vagrant

Terraform is one of the choices.